src/debugger/direct/directDebugAdapter.ts

```diff
@@ -107,32 +111,36 @@ export class DirectDebugAdapter extends ChromeDebugAdapter {
 
         return new Promise<void>((resolve, reject) => this.initializeSettings(attachArgs)
             .then(() => {
-                this.outputLogger("Attaching to the app");
-                logger.verbose(`Attaching to app: ${JSON.stringify(attachArgs, null , 2)}`);
-                return TelemetryHelper.generate("attach", extProps, (generator) => {
-                    return this.remoteExtension.getPackagerPort(attachArgs.cwd)
-                        .then((packagerPort: number) => {
-                            attachArgs.port = attachArgs.port || packagerPort;
-                            this.outputLogger(`Connecting to ${attachArgs.port} port`);
-                            const attachArguments = Object.assign({}, attachArgs, {
-                                address: "localhost",
-                                port: attachArgs.port,
-                                restart: true,
-                                request: "attach",
-                                remoteRoot: undefined,
-                                localRoot: undefined,
-                            });
-                            super.attach(attachArguments).then(() => {
-                                this.outputLogger("The debugger attached successfully");
-                                resolve();
-                            }).catch((e) => reject(e));
-                        }).catch((e) => reject(e));
-            })
-            .catch((err) => {
-                this.outputLogger("An error occurred while attaching to the debugger. " + err.message || err, true);
-                this.cleanUp();
-                reject(err);
-            });
+                this.outputLogger("Attaching to the application");
+                logger.verbose(`Attaching to the application: ${JSON.stringify(attachArgs, null , 2)}`);
+                return ReactNativeProjectHelper.getReactNativeVersionFromProjectPackage(attachArgs.cwd)
```

Comment:
I replaced
```
return ReactNativeProjectHelper.getReactNativeVersionFromProjectPackage(launchArgs.cwd)
```
with
```
return ReactNativeProjectHelper.getReactNativeVersion(attachArgs.cwd)
```
I did so, because there was an error handling problem when 
```
return ReactNativeProjectHelper.getReactNativePackageVersionFromNodeModules(launchArgs.cwd)
```
function was used here.